src/Core/Services/OrganizationService.cs

```
@@ -63,13 +63,25 @@ public async Task ClearAllAsync(string userId)
 
         public async Task<OrganizationDomainSsoDetailsResponse> GetClaimedOrganizationDomainAsync(string userEmail)
         {
-            if (string.IsNullOrEmpty(userEmail))
+            try
             {
-                return null;
+                if (string.IsNullOrEmpty(userEmail))
+                {
+                    return null;
+                }
+
+                return await _apiService.GetOrgDomainSsoDetailsAsync(userEmail);
             }
+            catch (Exception ex)
+            {
+                if (ex is ApiException apiEx && apiEx.Error != null && apiEx.Error.StatusCode == HttpStatusCode.NotFound)
+                {
+                    // this is a valid case so there is no need to show an error
+                    return null;
+                }
 
-            var response = await _apiService.GetOrgDomainSsoDetailsAsync(userEmail);
-            return response;
+                throw ex;
+            }
```

You don't need to catch all exception to do this, you can just write it as:
```c#
catch (ApiException ex) when (apiEx.Error?.StatusCode == HttpStatusCode.NotFound)
{
    // this is a valid case so there is no need to show an error
    return null;
}
```